model/Service/EncryptionSymmetricServiceHelper.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoEncryption\Service;
+
+use oat\taoEncryption\Service\KeyProvider\SimpleKeyProviderService;
+
+trait EncryptionSymmetricServiceHelper
+{
+    /** @var SimpleKeyProviderService */
+    private $encryptionService;
+
+    /**
+     * @param string $key
+     * @return EncryptionSymmetricService
+     * @throws \Exception
+     */
+    public function getEncryptionService($key)
+    {
+        if (is_null($this->encryptionService)) {
+            /** @var EncryptionSymmetricService $service */
+            $service = $this->getServiceLocator()->get($this->getOption(static::OPTION_ENCRYPTION_SERVICE));
```

Comment:
This sound like a magic for me, do u want u to replace it by an abstract method?